nf_core/lint_utils.py

```diff
@@ -60,12 +63,55 @@ def run_prettier_on_file(file):
         If Prettier is not installed, a warning is logged.
     """
 
+    if shutil.which("prettier"):
+        _run_prettier_on_file(file)
+    elif shutil.which("pre-commit"):
+        _run_pre_commit_prettier_on_file(file)
+    else:
+        log.warning(
+            "Neither Prettier nor the prettier pre-commit hook are available. At least one of them is required."
+        )
+
+
+def _run_prettier_on_file(file):
+    """Run natively installed Prettier on a file."""
+
     try:
         subprocess.run(
             ["prettier", "--write", file],
-            stdout=subprocess.DEVNULL,
-            stderr=subprocess.DEVNULL,
-            check=False,
+            capture_output=True,
+            check=True,
+        )
+    except subprocess.CalledProcessError as e:
+        if ": SyntaxError: " in e.stderr.decode():
+            raise ValueError(f"Can't format {file} because it has a synthax error.\n{e.stderr.decode()}") from e
+        raise ValueError(
+            "There was an error running the prettier pre-commit hook.\n"
+            f"STDOUT: {e.stdout.decode()}\nSTDERR: {e.stderr.decode()}"
+        ) from e
```

Comment:
> And include pre-commit as a dependency in tools?

This PR makes pre-commit a dependency. Installing the hooks is not necessary, they can be run directly. Installing them with `pre-commit install` just makes them run before any commit in the repo. So install is not (just) installing the tools but setting a git repo up to use them. `pre-commit run` will also install the hooks but not change any settings in any git repo.

The way it is set up means the user doesn't need to handle any installation manually and neither is there a need for a separate `nf-core lint setup` command.

Reply:
Okay, I'm following now. Still wondering, since it'll be installed, why allow the system `prettier` install? I get it's probably faster, less stuff installed, but more code for us to maintain.

Reply:
I am all for only using prettier via the pre-commit hook mechanism. The global Prettier would make the first invocation a bit quicker, because pre-commit has to pull the hook from the web the first time it runs. I also have some questions about the configuration for when prettier is invoked from a global install, because it might not see the same configs as the one from the pre-commit hooks. I'd have to look at this, too. 
One more point for only using pre-commit prettier is that the code - both n production and testing - would become much simpler.

Reply:
I am tempted to actually promote running pre-commit install by default. It would make the developer experience more platform independent, but it also installs pre-commit hooks in all repos nf-core is used in and this may or may not be desirable depending on the project.

Reply:
> because pre-commit has to pull the hook from the web the first time it runs.

And this also menas internet access is necessary for the first time prettier is run. If a developer doesn't have internet access but prettier is installed on the machine it would help to keep the current setup, although it is more complex. (hitherto I remain ignorant about the issue of the configuration for Prettier native)

Reply:
I'd like to move the discussion about the use of pre-commit install out of this PR because I see it as a separate issue that does not decrease the usefulness of this PR, but considerably complicates it and would drag the merge of this PR unnecessarily.

Reply:
How/where does running `pre-commit prettier` get it's config?  Although the primary use-case for tools is working with the public `nf-core` repos I am mindful of the fact that people do use them internally (selfish bias here too as I do) and wonder whether there are cases where running `prettier` (or using some fixed config) is undesirable?

If `pre-commit prettier` draws its config from the repo/global config files in the same way as running manually installed prettier does then this is probably less of an issue

Reply:
The hooks take the configuration in the repository where they are run into account.